src/relay/backend/aot_executor_codegen.cc

```diff
@@ -1063,6 +1064,8 @@ class AOTExecutorCodegen : public MixedModeVisitor {
     }
 
     mod = transform::ToANormalForm()(mod);
+    mod = transform::InferType()(mod);
+    mod = transform::AnnotateUsedMemory()(mod);
```

Comment:
how come you want to do this at the relay level when we would further analyze this in USMP down below?

Reply:
This is mainly to get an idea how aggressive the scheduling need to be in-terms of memory when we lower them to TIR.

Yes, USMP will analyze and optimize further post-scheduling attempting to be hit "a" theoretical minimum (i.e. memory pressure).

Reply:
got it, so is this intended as cascader input?

Reply:
Yes, the cascader is the usecase we are interested in.